plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -93,6 +93,8 @@ func init() {
 	factory.RegisterFitPredicate("HostName", predicates.PodFitsHost)
 	// Fit is determined by node selector query.
 	factory.RegisterFitPredicate("MatchNodeSelector", predicates.PodSelectorMatches)
+	// Optional, cluster-autoscaler friendly priority function - give used nodes higher priority.
+	factory.RegisterPriorityFunction("MostRequestedPriority", priorities.MostRequestedPriority, 1)
```

Comment:
OK - so basically this PR in the current shape is a no-op (and it's intended to be).
The next PR will do the real work, once we already have the new function registered.

>
We can enable it using scheduler config file, but we cannot disable LeastRequestedPriority,

Why? I looked into the code and it seems that when you define the policy-config, then it doesn't use any default predicates/priorities - it only uses the ones defined in you policy config.

Anyway - I think that the desired solution is slightly different. Currently we have something called "DefaultAlgorithmProvider" which is using default predicates and priorities.
How I would see the solution is to create a new algorithm provider (ClusterAutoscalerAlgorithmProvider (probably we need a better name) that would differ from DefaultAlgorithmProvider.

Then, (when cluster autoscaler is switched on) we will set "--algorithm-provider=ClusterAutoscalerAlgorithmProvider" flag on the scheduler, and it will just be working.

Am I missing something?